### ISSUE TITLE
Error importing json into Foundry

### ISSUE THREAD

**Issue by S1thlord597** [qrizzl/ed4-steps2foundry#6]
I am seeing this error when importing the json. Foundry version 12 build 331.

Getting an error importing json. It is importing some data, like some talents, Discipline, race. Not attributes, Skills equipment.

![image](https://github.com/user-attachments/assets/6f708250-116c-4a2f-a17d-cdeb8315a081)

**Comment by qrizzl** [comment]
Hei, thanks for the report. Is it possible for you to upload the json file you where trying to import?

**Comment by S1thlord597** [comment]
Yeah no problems, I have also included some screenshots from the sheets within Foundry. Notably, attributes do not get imported, talent ranks do not get imported. In this example, most equipment but notably the ammunition isn't imported.

Let me know if you need something else.
[test_import.json](https://github.com/user-attachments/files/17668661/test_import.json)
![Imported_sheet3](https://github.com/user-attachments/assets/f86def02-1863-48e7-8984-7231e9b4228b)
![Steps_equipment](https://github.com/user-attachments/assets/fee8083d-baf1-4194-a699-2d2c24551d94)
![Steps2foundry_version](https://github.com/user-attachments/assets/d9abaaf5-d9a2-4eb4-ad88-5e51c3008e6b)
![Imported_sheet1](https://github.com/user-attachments/assets/e540faef-1dc7-4dee-8244-d9b01c5274f5)
![Imported_sheet2](https://github.com/user-attachments/assets/f9be7eed-26e4-4575-932c-b89d68c01c28)

**Comment by qrizzl** [comment]
The missing equipment is expected since it's a very sketchy way of trying to find possible matches in the compendia.
For the talent ranks: I was unfortunately not able to reproduce the error.
Attributes: that might actually have to do with ED version and a change in the construction of the character sheet.

Was this gonna be a single time import? Or are you planning on doing it on a frequent basis? In the first case I would suggest to adjust things manually, since it might take some time for me to get a chance to fix it.  I'm also trying to wait with any changes to this module since we're currently working on some major changes in the ED system, which will then entail a rework (and hopefully improvement) of this module as well. But that might take a bit of time yet.

**Comment by S1thlord597** [comment]
Hi Chris,

Thanks for having a look and letting me know.
As for equipment, not a huge deal, as I have only seen this with ammo atm so can be added with little fuss.

I think the talent issue may have been related to the note you made about reloading the sheet the first time. I saw your notes and did a fresh import of an updated sheet, reloaded and say the talents and ranks.

Attributes are a pain but as you say, with what you're waiting for, that all makes sense.

Am I correct in the impression that you're also part of the Fasa team? I believe that Fasa have their own resources working on Foundry too. If so that's awesome, I thought you were an independent. Either way the work is awesome and I really appreciate it. I am in Australia and without foundry or VTT's there would be no way I could play ED. I play with my old crew who now live all around the world.

I love ED, played it since '97 and just started to get my old crew back into it and rediscover my love of it and I plan on driving new interest in the game off the back of the post covid d&d wave. I think ED would be really popular but has always lacked real marketing, which in this day and age is a lot easier for the average no body to contribute to.

All that is to answer, yes I plan on many imports from secondstep as I find it a robust, faster and mechanically more trustworthily calculation of the character creation rules set than what is currently in Foundry.

The thing I would love more than any is a random character generator, so I could build out npc's and enemies quickly. Something you could say "I want a C3 Archer" or "A non-adept C2 Fighter" and it would build out one, randomise the race if you don't specify and fill in all stats and equipment.

I have tried this with AI at the moment, but it invents things that are not correct, like a Talent names that don't exist or are really from d&d and then the calculated ranks and base stats are all wrong.

But still without that generator secondstep is a faster character builder so I do rely on it.

**Comment by qrizzl** [comment]
Thank you so much, that's lovely to hear it helps. :) 

First: no I'm not part of the Fasa Team, just like Earthdawn and Coding ^^ I am involved in developing the system though, that's why I know what is planned and coming up.

It's really great to hear these stories, they seem familiar too. I do think Earthdawn lends itself really well to a Virtual Tabletop since it's definitely a little more complex than one dice system or others, but that's what I love about it. And I hope we can help giving even more of the love it deserves through the Foundry system.

The idea is of a random char generator is great, and I'm sure Patrick would love to take it into consideration for new features. Definitely post that on the Discord, or in the github project!

For now, I hope we can get to a new version very fast, then this module will be updated accordingly and improve a lot too!

**Comment by S1thlord597** [comment]
Thanks very much for your response. I'll definitely post that idea for a generator up.

Appreciate your time and effort :)

Sent with [Proton Mail](https://proton.me/mail/home) secure email.
